.github/workflows/ncbench.yml

```diff
@@ -0,0 +1,116 @@
+name: NCBench Zenodo Upload
+# This workflow can be triggered manually with the GitHub actions workflow dispatch button.
+
+on:
+  # upload can only be triggered manually for now
+  workflow_dispatch:
+    inputs:
+      germline_default_agilent:
+        description: "Trigger NCBench upload"
+        type: boolean
+        default: true
+  pull_request: # TODO: remove once it works
+
+jobs:
+  ncbench-upload:
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          # TODO: Add in other test profiles
+          - profile: germline_default_agilent
+            enabled: ${{ ( github.repository == 'nf-core/sarek' ) && ( github.event_name != 'workflow_dispatch' || inputs.germline_default_agilent ) }}
+    steps:
+      - name: Download pipeline
+        uses: actions/checkout@v4
+
+      - name: Extract Versions
+        id: extract_versions
+        run: |
+          PIPELINE_VERSION=$(awk -F"'" '/manifest {/,/version/ {if(/version/) print $2}' nextflow.config)
+          echo "PIPELINE_VERSION=${PIPELINE_VERSION}" >> $GITHUB_ENV
+
+          echo "PIPELINE_VERSION_NO_DOTS=$(echo $PIPELINE_VERSION | tr -d '.')" >> $GITHUB_ENV
+
+          FASTP_VERSION=$(awk '/- bioconda::fastp=/ {gsub(/.*=/, "", $NF); gsub(/'\''/, "", $NF); print $NF}' modules/nf-core/fastp/environment.yml)
+          echo "FASTP_VERSION=${FASTP_VERSION}" >> $GITHUB_ENV
+
+          BWA_VERSION=$(awk '/- bwa=/ {gsub(/.*=/, "", $NF); gsub(/'\''/, "", $NF); print $NF}' modules/nf-core/bwa/mem/environment.yml)
+          echo "BWA_VERSION=${BWA_VERSION}" >> $GITHUB_ENV
+
+          BQSR_VERSION=$(awk '/- bioconda::gatk4=/ {gsub(/.*=/, "", $NF); gsub(/'\''/, "", $NF); print $NF}' modules/nf-core/gatk4/baserecalibrator/environment.yml)
+          echo "BQSR_VERSION=${BQSR_VERSION}" >> $GITHUB_ENV
+
+          DEEPVARIANT_VERSION=$(awk -F':' '/container/ {gsub(/"/, "", $2); print $2}' modules/nf-core/deepvariant/main.nf)
+          echo "DEEPVARIANT_VERSION=${DEEPVARIANT_VERSION}" >> $GITHUB_ENV
+
+          FREEBAYES_VERSION=$(awk '/- bioconda::freebayes=/ {gsub(/.*=/, "", $NF); gsub(/'\''/, "", $NF); print $NF}' modules/nf-core/freebayes/environment.yml)
+          echo "FREEBAYES_VERSION=${FREEBAYES_VERSION}" >> $GITHUB_ENV
+
+          HAPLOTYPECALLER_VERSION=$(awk '/- bioconda::gatk4=/ {gsub(/.*=/, "", $NF); gsub(/'\''/, "", $NF); print $NF}' modules/nf-core/gatk4/haplotypecaller/environment.yml)
+          echo "HAPLOTYPECALLER_VERSION=${HAPLOTYPECALLER_VERSION}" >> $GITHUB_ENV
+
+          STRELKA_VERSION=$(awk '/- bioconda::strelka=/ {gsub(/.*=/, "", $NF); gsub(/'\''/, "", $NF); print $NF}' modules/nf-core/strelka/germline/environment.yml)
+          echo "STRELKA_VERSION=${STRELKA_VERSION}" >> $GITHUB_ENV
```

Comment:
This could get moved into a script if you want for readability, and not needing any escapes.

But it might break the $GITHUB_ENV? IDK

Reply:
I thin kit might break the GITHUB ENV stuff. at least with the python script I had to write it to file and then read it out again. Might still look cleaner though. i am taking a look at the jq stuff Adam mentioned